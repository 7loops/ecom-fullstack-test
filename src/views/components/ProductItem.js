import React from "react";
import ProductHeader from "./ProductHeader";
import ProductPrice from "./ProductPrice";
import ProductCta from "./ProductCta";

const ProductItem = props => {
  return (
    <div className="product">
      <article className="product-item">
        <ProductHeader product={props.product} />
        <div className="product-body">
          <h2 className="product-title">
            <a href={props.product.ctaLink}>{props.product.title}</a>
          </h2>
          <p className="product-description">{props.product.description}</p>
          <div>
            <ProductPrice product={props.product} />
            <ProductCta product={props.product} />
          </div>
        </div>
        <div />
      </article>
    </div>
  );
};

export default ProductItem;
