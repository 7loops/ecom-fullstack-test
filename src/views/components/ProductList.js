import React from "react";
import { connect } from "react-redux";
import ProductItem from "./ProductItem";

const renderItems = products => {
  if (products) {
    return products.map(renderItem);
  }
};

const renderItem = (product, i) => {
  return <ProductItem key={i} product={product} />;
};

export const ProductList = props => {
  return <div className="products-list">{renderItems(props.products)}</div>;
};

const mapStateToProps = state => {
  return {
    products: state.productReducer
  };
};

export default connect(mapStateToProps)(ProductList);
