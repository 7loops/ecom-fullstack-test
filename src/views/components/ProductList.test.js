import React from "react";
import { mount } from "enzyme";
import ProductItem from "./ProductItem";
import { ProductList } from "./ProductList";

import storeProducts from "../../constants/products";

describe("<ProductList />", () => {
  it("Product list should be empty", () => {
    const container = mount(<ProductList products={storeProducts} />);
    expect(container.find(ProductItem).length).toBe(2);
  });

  it("Product list should be equal to 2", () => {
    const container = mount(<ProductList products={storeProducts} />);
    expect(container.find(ProductItem).length).toBe(2);
  });

  it("Product list should match snapshot", () => {
    const container = mount(<ProductList products={storeProducts} />);
    expect(container).toMatchSnapshot();
  });
});
