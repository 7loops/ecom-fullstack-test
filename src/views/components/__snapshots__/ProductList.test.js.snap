// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> Product list should match snapshot 1`] = `
<ProductList
  products={
    Array [
      Object {
        "cta": "Shop Now",
        "ctaLink": "/random/link/to/no/where",
        "currency": "£",
        "description": "Lets your pictures speak for themselves.",
        "image": Object {
          "alt": "Simple Canvas",
          "path": "/images/product.jpg",
        },
        "price": 1500,
        "priceLabel": "From",
        "productLabel": "bestseller",
        "title": "Simple Canvas",
      },
      Object {
        "cta": "Shop Now",
        "ctaLink": "/random/link/to/no/where",
        "currency": "£",
        "description": "Can't choose just one pic? Put your favourite photos on one canvas - a collage.",
        "image": Object {
          "alt": "Collage Canvas",
          "path": "/images/product.jpg",
        },
        "price": 2500,
        "priceLabel": "From",
        "productLabel": "",
        "title": "Collage Canvas",
      },
    ]
  }
>
  <div
    className="products-list"
  >
    <ProductItem
      product={
        Object {
          "cta": "Shop Now",
          "ctaLink": "/random/link/to/no/where",
          "currency": "£",
          "description": "Lets your pictures speak for themselves.",
          "image": Object {
            "alt": "Simple Canvas",
            "path": "/images/product.jpg",
          },
          "price": 1500,
          "priceLabel": "From",
          "productLabel": "bestseller",
          "title": "Simple Canvas",
        }
      }
    >
      <div
        className="product"
      >
        <article
          className="product-item"
        >
          <ProductHeader
            product={
              Object {
                "cta": "Shop Now",
                "ctaLink": "/random/link/to/no/where",
                "currency": "£",
                "description": "Lets your pictures speak for themselves.",
                "image": Object {
                  "alt": "Simple Canvas",
                  "path": "/images/product.jpg",
                },
                "price": 1500,
                "priceLabel": "From",
                "productLabel": "bestseller",
                "title": "Simple Canvas",
              }
            }
          >
            <header
              className="product-header"
            >
              <span
                className="product-label"
              >
                bestseller
              </span>
              <a
                href="/random/link/to/no/where"
              >
                <picture
                  className="product-image"
                >
                  <img
                    alt="Simple Canvas"
                    src="/images/product.jpg"
                  />
                </picture>
              </a>
            </header>
          </ProductHeader>
          <div
            className="product-body"
          >
            <h2
              className="product-title"
            >
              <a
                href="/random/link/to/no/where"
              >
                Simple Canvas
              </a>
            </h2>
            <p
              className="product-description"
            >
              Lets your pictures speak for themselves.
            </p>
            <div>
              <ProductPrice
                product={
                  Object {
                    "cta": "Shop Now",
                    "ctaLink": "/random/link/to/no/where",
                    "currency": "£",
                    "description": "Lets your pictures speak for themselves.",
                    "image": Object {
                      "alt": "Simple Canvas",
                      "path": "/images/product.jpg",
                    },
                    "price": 1500,
                    "priceLabel": "From",
                    "productLabel": "bestseller",
                    "title": "Simple Canvas",
                  }
                }
              >
                <p
                  className="product-priceContainer"
                >
                  <span
                    className="product-priceLabel"
                  >
                    From
                     
                  </span>
                  <span
                    className="product-price"
                  >
                    £
                    1500
                  </span>
                </p>
              </ProductPrice>
              <ProductCta
                product={
                  Object {
                    "cta": "Shop Now",
                    "ctaLink": "/random/link/to/no/where",
                    "currency": "£",
                    "description": "Lets your pictures speak for themselves.",
                    "image": Object {
                      "alt": "Simple Canvas",
                      "path": "/images/product.jpg",
                    },
                    "price": 1500,
                    "priceLabel": "From",
                    "productLabel": "bestseller",
                    "title": "Simple Canvas",
                  }
                }
              >
                <a
                  className="product-cta"
                  href="/random/link/to/no/where"
                >
                  Shop Now
                </a>
              </ProductCta>
            </div>
          </div>
          <div />
        </article>
      </div>
    </ProductItem>
    <ProductItem
      product={
        Object {
          "cta": "Shop Now",
          "ctaLink": "/random/link/to/no/where",
          "currency": "£",
          "description": "Can't choose just one pic? Put your favourite photos on one canvas - a collage.",
          "image": Object {
            "alt": "Collage Canvas",
            "path": "/images/product.jpg",
          },
          "price": 2500,
          "priceLabel": "From",
          "productLabel": "",
          "title": "Collage Canvas",
        }
      }
    >
      <div
        className="product"
      >
        <article
          className="product-item"
        >
          <ProductHeader
            product={
              Object {
                "cta": "Shop Now",
                "ctaLink": "/random/link/to/no/where",
                "currency": "£",
                "description": "Can't choose just one pic? Put your favourite photos on one canvas - a collage.",
                "image": Object {
                  "alt": "Collage Canvas",
                  "path": "/images/product.jpg",
                },
                "price": 2500,
                "priceLabel": "From",
                "productLabel": "",
                "title": "Collage Canvas",
              }
            }
          >
            <header
              className="product-header"
            >
              <a
                href="/random/link/to/no/where"
              >
                <picture
                  className="product-image"
                >
                  <img
                    alt="Collage Canvas"
                    src="/images/product.jpg"
                  />
                </picture>
              </a>
            </header>
          </ProductHeader>
          <div
            className="product-body"
          >
            <h2
              className="product-title"
            >
              <a
                href="/random/link/to/no/where"
              >
                Collage Canvas
              </a>
            </h2>
            <p
              className="product-description"
            >
              Can't choose just one pic? Put your favourite photos on one canvas - a collage.
            </p>
            <div>
              <ProductPrice
                product={
                  Object {
                    "cta": "Shop Now",
                    "ctaLink": "/random/link/to/no/where",
                    "currency": "£",
                    "description": "Can't choose just one pic? Put your favourite photos on one canvas - a collage.",
                    "image": Object {
                      "alt": "Collage Canvas",
                      "path": "/images/product.jpg",
                    },
                    "price": 2500,
                    "priceLabel": "From",
                    "productLabel": "",
                    "title": "Collage Canvas",
                  }
                }
              >
                <p
                  className="product-priceContainer"
                >
                  <span
                    className="product-priceLabel"
                  >
                    From
                     
                  </span>
                  <span
                    className="product-price"
                  >
                    £
                    2500
                  </span>
                </p>
              </ProductPrice>
              <ProductCta
                product={
                  Object {
                    "cta": "Shop Now",
                    "ctaLink": "/random/link/to/no/where",
                    "currency": "£",
                    "description": "Can't choose just one pic? Put your favourite photos on one canvas - a collage.",
                    "image": Object {
                      "alt": "Collage Canvas",
                      "path": "/images/product.jpg",
                    },
                    "price": 2500,
                    "priceLabel": "From",
                    "productLabel": "",
                    "title": "Collage Canvas",
                  }
                }
              >
                <a
                  className="product-cta"
                  href="/random/link/to/no/where"
                >
                  Shop Now
                </a>
              </ProductCta>
            </div>
          </div>
          <div />
        </article>
      </div>
    </ProductItem>
  </div>
</ProductList>
`;
